lib/ansible/parsing/dataloader.py

```diff
@@ -273,9 +274,10 @@ def path_dwim_relative(self, path, dirname, source, is_role=False):
         '''
 
         search = []
+        source = to_text(source, errors='surrogate_or_strict')
 
         # I have full path, nothing else needs to be looked at
-        if source.startswith('~') or source.startswith(os.path.sep):
+        if RE_ABS_PATH.search(source):
```

Comment:
RE_ABS_PATH is maybe a little too concrete. Maybe wrap in a is_abs_path() method? ditto for RE_TASKS